src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1563,21 +1563,40 @@ abstract class RefChecks extends Transform {
 
           transform(qual)
       case StringContextIntrinsic(treated, args) =>
-        var result: Tree = treated.head
-        def concat(t: Tree): Unit = {
-          result = atPos(t.pos)(gen.mkMethodCall(gen.mkAttributedSelect(result, definitions.String_+), t :: Nil)).setType(StringTpe)
-        }
+        val argsIndexed = args.toVector
+        var concatArgs = ListBuffer[Tree]()
         val numLits = treated.length
         foreachWithIndex(treated.tail) { (lit, i) =>
           val treatedContents = lit.asInstanceOf[Literal].value.stringValue
           val emptyLit = treatedContents.isEmpty
           if (i < numLits - 1) {
-            concat(args(i))
-            if (!emptyLit) concat(lit)
+            concatArgs += argsIndexed(i)
+            if (!emptyLit) concatArgs += lit
           } else if (!emptyLit) {
-            concat(lit)
+            concatArgs += lit
           }
         }
+        def mkConcat(pos: Position, lhs: Tree, rhs: Tree): Tree =
+          atPos(pos)(gen.mkMethodCall(gen.mkAttributedSelect(lhs, definitions.String_+), rhs :: Nil)).setType(StringTpe)
+
+        var result: Tree = treated.head
+        val list = concatArgs.toList.grouped(32).toList
```

Comment:
I can see a couple of improvements here: 
- Instead of directly going to the `concatArgs.toList.grouped`, perhaps we can have a top-level `if-then-else` using `lengthCompare` to 32.
- If necessary, for performance: in the _chunked_ branch (more than 32), instead of using `grouped(32)` and then the `foreach`, we could have an iterative outer loop that, at each iteration, uses the first 32 and then continues with the rest. If this is using a `List`, you can have two nested loops, which share (and advance) the same variable.